algoliasearch/client_test.go

```diff
@@ -67,16 +67,22 @@ func hasString(slice []string, elt string) bool {
 	return false
 }
 
-func checkNbHits(got, expected int, t *testing.T) {
-	checkEqual(got, expected, "nbHits", t)
+func checkNbHits(got, expected int, index Index, t *testing.T) {
+	checkEqual(got, expected, "nbHits", index, t)
 }
 
-func checkEqual(got, expected interface{}, message string, t *testing.T) {
+func checkEqual(got, expected interface{}, message string, index Index, t *testing.T) {
 	if got != expected {
-		t.Fatalf("%s: %v expected %v", message, got, expected)
+		fatal(index, t, "%s: %v expected %v", message, got, expected)
 	}
 }
 
+func fatal(index Index, t *testing.T, fmt string, objs ...interface{}) {
```

Comment:
For all the functions that require the `*testing.T`, I would suggest to put it first in the parameter list. Looks more Go-ish IMHO.